File: New folder/cdp-sdk-8fcfac348487f586b224b77074f52594ad44198b/examples/python/solana/accounts/export_account.py
# Usage: uv run python solana/accounts/export_account.py

import asyncio
import base58
from cdp import CdpClient
from dotenv import load_dotenv

load_dotenv()


async def main():
    async with CdpClient() as cdp:
        # Get or create account
        account = await cdp.solana.get_or_create_account(
            name="MyAccount",
        )
        print("Account: ", account.address)

        # Exporting account by address
        print("--------------------------------")
        print("Exporting account by address...")
        exported_private_key_by_address = await cdp.solana.export_account(
            address=account.address
        )
        print("Exported private key: ", exported_private_key_by_address)

        # Verify public key (last 32 bytes are the public key)
        full_key_bytes_by_address = base58.b58decode(exported_private_key_by_address)
        public_key_bytes_by_address = full_key_bytes_by_address[32:]
        public_key_by_address = base58.b58encode(public_key_bytes_by_address).decode('utf-8')
        print("Public key derived from private key:", public_key_by_address)

        # Exporting account by name
        print("--------------------------------")
        print("Exporting account by name...")
        exported_private_key_by_name = await cdp.solana.export_account(
            name="MyAccount"
        )
        print("Exported private key: ", exported_private_key_by_name)

        # Verify public key (last 32 bytes are the public key)
        full_key_bytes_by_name = base58.b58decode(exported_private_key_by_name)
        public_key_bytes_by_name = full_key_bytes_by_name[32:]
        public_key_by_name = base58.b58encode(public_key_bytes_by_name).decode('utf-8')
        print("Public key derived from private key:", public_key_by_name)


asyncio.run(main())

File: New folder/cdp-sdk-8fcfac348487f586b224b77074f52594ad44198b/examples/typescript/solana/accounts/exportAccount.ts
// Usage: pnpm tsx solana/accounts/exportAccount.ts

import { CdpClient } from "@coinbase/cdp-sdk";
import bs58 from "bs58";
import "dotenv/config";

const cdp = new CdpClient();

const account = await cdp.solana.getOrCreateAccount({
  name: "MyAccount",
});
console.log("Account:", account.address);

// Exporting account by address
console.log("--------------------------------");
console.log("Exporting account by address...");
const exportedPrivateKeyByAddress = await cdp.solana.exportAccount({
  address: account.address,
});
console.log("Exported private key:", exportedPrivateKeyByAddress);

// Verify public key (last 32 bytes are the public key)
const fullKeyBytesByAddress = bs58.decode(exportedPrivateKeyByAddress);
const publicKeyBytesByAddress = fullKeyBytesByAddress.subarray(32);
const derivedPublicKeyByAddress = bs58.encode(publicKeyBytesByAddress);
console.log("Public key derived from private key:", derivedPublicKeyByAddress);

// Exporting account by name
console.log("--------------------------------");
console.log("Exporting account by name...");
const exportedPrivateKeyByName = await cdp.solana.exportAccount({
  name: "MyAccount",
});
console.log("Exported private key:", exportedPrivateKeyByName);

// Verify public key (last 32 bytes are the public key)
const fullKeyBytesByName = bs58.decode(exportedPrivateKeyByName);
const publicKeyBytesByName = fullKeyBytesByName.subarray(32);
const derivedPublicKeyByName = bs58.encode(publicKeyBytesByName);
console.log("Public key derived from private key:", derivedPublicKeyByName);
